src/DeepCopy/DeepCopy.php

```diff
@@ -141,6 +141,10 @@ private function copyObject($object)
         $newObject = clone $object;
         $this->hashMap[$objectHash] = $newObject;
 
+        if (!$reflectedObject->isUserDefined()) {
```

Comment:
Tests needed. The tests should basically justify the "why" of this change. Please be very detailed with test assets, including internal classes, non-internal classes, classes extending internal classes, and mixed object trees of both.